boot/boot_images.cmake

```diff
@@ -59,6 +61,53 @@ ${_efisys_file} 1
 set(ISO_MANUFACTURER "ReactOS Foundation")  # For both the publisher and the preparer
 set(ISO_VOLNAME      "ReactOS")             # For both the Volume ID and the Volume set ID
 
+# ISO image options
+set(ISO_COMMON_OPTIONS
+    -iso-level 4 -publisher ${ISO_MANUFACTURER} -preparer ${ISO_MANUFACTURER} -volid ${ISO_VOLNAME} -volset ${ISO_VOLNAME})
+
+set(ISO_LAYOUT_OPTIONS
+    -duplicates-once -no-cache-inodes -graft-points)
+
+## "El Torito" ISO boot options.
+#
+set(ISO_BOOT_OPTIONS)
+set(ISO_BOOT_REGTEST_OPTIONS) # Useful only for BootCDRegTest
+
+# BIOS-based PC boot entry (x86/x64 only)
+if(ARCH STREQUAL "i386" OR ARCH STREQUAL "amd64")
+    set(ISO_BOOT_OPTIONS
+        -eltorito-platform x86 -eltorito-boot loader/isoboot.bin -no-emul-boot -boot-load-size 4)
+    set(ISO_BOOT_REGTEST_OPTIONS
+        -eltorito-platform x86 -eltorito-boot loader/isobtrt.bin -no-emul-boot -boot-load-size 4)
+endif()
+
+# EFI boot entry
+list(APPEND EFI_SUPPORTED_ARCHS "i386" "amd64" "ia64" "arm" "arm64")
+if(ARCH IN_LIST EFI_SUPPORTED_ARCHS)
+    set(ISO_BOOT_EFI_OPTIONS
+        -eltorito-platform efi -eltorito-boot loader/efisys.bin -no-emul-boot)
+else()
+    message(WARNING "Unknown ARCH '" ${ARCH} "', cannot add UEFI boot support to the ISO image.")
+endif()
+
+# Boot common options (boot catalog, boot files sorting)
+set(ISO_BOOT_COMMON_OPTIONS
+    -hide boot.catalog -sort ${CMAKE_CURRENT_BINARY_DIR}/bootfiles.sort)
+
+# Construct the resulting boot options
+if(ARCH STREQUAL "i386" OR ARCH STREQUAL "amd64")
+    set(ISO_BOOT_OPTIONS ${ISO_BOOT_OPTIONS}
+        -eltorito-alt-boot ${ISO_BOOT_EFI_OPTIONS}
+        ${ISO_BOOT_COMMON_OPTIONS})
+
+    set(ISO_BOOT_REGTEST_OPTIONS ${ISO_BOOT_REGTEST_OPTIONS}
+        -eltorito-alt-boot ${ISO_BOOT_EFI_OPTIONS}
+        ${ISO_BOOT_COMMON_OPTIONS})
+else()
+    set(ISO_BOOT_OPTIONS ${ISO_BOOT_EFI_OPTIONS} ${ISO_BOOT_COMMON_OPTIONS})
```

Comment:
I don't think it makes sense to create an ISO image for non-x86 architectures. Better generate fat32 image for them right away

Reply:
They make sense for VMs and hardware with ACPI/UEFI, They're easier to debug with and say someone wanted to deploy our image on a surface RT for instance or even the dev kit Microsoft sells.

ISOs make sense and they save a lot of hassle 

I understand for making machines without UEFI/ACPI just using a fat32 image

but our generic ISOs for arm32 - arm64 need to be ISOs